controllers/policy_utils.go

```diff
@@ -105,6 +105,16 @@ func reconcilePolicy(ctx context.Context, client client.Client, reconciler admis
 	}
 
 	if !isPolicyUniquelyReachable(ctx, client, &policyServerDeployment) {
+		// we need to check if the policy has status active because
+		// when installing a new policy, the first reconciliation loop may
+		// occur before the new replicaset creation. Which means that
+		// the comparisons in the isPolicyUniquelyReachable returns the policy
+		// as unique reachable. Which is true in that point in time.
+		// Therefore, the policy is set as "active". In the second
+		// reconciliation, we need to fix that.
+		if policy.GetStatus().PolicyStatus == v1alpha2.PolicyStatusActive {
+			policy.SetStatus(v1alpha2.PolicyStatusPending)
+		}
```

Comment:
I think it would be better to avoid setting its status to active when no replica set exists yet, so that we don't have to maintain side fixes. For example, this change would imply that a user watching the policy will see it going from `active` to `pending`, being `active` for a very short period of time. I think we should make this impossible in the first place.

Reply:
yes, I agree. I would not set the status to active if the replica set is not created yet. Looking into the code it should return false if the replica set is not there https://github.com/jvanz/kubewarden-controller/blob/46827eaf2fa1664721d7f4a94b783966dbdcf24d/controllers/policystatus_utils.go#L93-L94 we should investigate why this is not working

Reply:
I agree that the `active` -> `pending` -> `active` is not nice. But the problem is that there is a `replicaset` when we install a policy. For example, the user installs the policy server with no policy yet. There is a replica set to the current policy server. Then, the user installs the policy. This triggers a reconciliation, and the policy status `reconciler` starts. When the reconcile loop checks if the `replicaset` is the latest it will returns true. Because the new `replicaset` used to deploy the policy is not there yet. Thus, the `replicaset` version will match with the version in the `deployment`. Returning the policy as unique reachable setting it as `active`. 

I could not find a way to detect that the current `replicaset`, even matching the `deployment` version, is not the latest `replicaset`.

Reply:
> I could not find a way to detect that the current replicaset, even matching the deployment version, is not the latest replicaset.

A possible way would be to ignore replicasets with a `.metadata.creationTimestamp` lower than the policy`.metadata.creationTimestamp`. This way, we know that it's not a replica set we should compare against.

Reply:
Never mind, I'm mixing the meanings of the status `active` and the condition `isPolicyUniquelyReachable`.